evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -42,6 +45,18 @@ def _make_entity_set(self, X):
             es = ft_es.add_dataframe(dataframe=X, dataframe_name="X", index=self.index)
         return es
 
+    def _filter_features(self, X, es):
+        features_to_use = []
+        for feature in self.features:
+            if not feature.entityset == es:
+                continue
+            if not isinstance(feature, IdentityFeature) and set(
```

Comment:
`not isinstance(feature, IdentityFeature)` is needed because by definition, if an `IdentityFeature` is passed in it will be part of `X`.